x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/ConnectorFiltering.java

```diff
@@ -0,0 +1,591 @@
+                return new FilteringRule(createdAt, field, id, order, policy, rule, updatedAt, value);
+            }
+        }
+    }
+
+    public static class FilteringValidation implements Writeable, ToXContentObject {
+        private final List<String> ids;
+        private final List<String> messages;
+
+        /**
+         * Constructs a new FilteringValidation instance.
+         *
+         * @param ids      The list of identifiers associated with the validation.
+         * @param messages The list of messages describing the validation results.
+         */
+        public FilteringValidation(List<String> ids, List<String> messages) {
+            this.ids = ids;
+            this.messages = messages;
+        }
+
+        public FilteringValidation(StreamInput in) throws IOException {
+            this.ids = in.readStringCollectionAsList();
+            this.messages = in.readStringCollectionAsList();
+        }
+
+        @Override
+        public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+            builder.startObject();
+            {
+                builder.stringListField("ids", ids);
+                builder.stringListField("messages", messages);
+            }
+            return builder;
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            out.writeStringCollection(ids);
+            out.writeStringCollection(messages);
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if (o == null || getClass() != o.getClass()) return false;
+            FilteringValidation that = (FilteringValidation) o;
+            return Objects.equals(ids, that.ids) && Objects.equals(messages, that.messages);
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(ids, messages);
+        }
+
+        public static class Builder {
+
+            private List<String> ids;
+            private List<String> messages;
+
+            public Builder setIds(List<String> ids) {
+                this.ids = ids;
+                return this;
+            }
+
+            public Builder setMessages(List<String> messages) {
+                this.messages = messages;
+                return this;
+            }
+
+            public FilteringValidation build() {
+                return new FilteringValidation(ids, messages);
+            }
+        }
+    }
+
+    public enum FilteringValidationState {
+        EDITED,
+        INVALID,
+        VALID;
+
+        @Override
+        public String toString() {
+            return name().toLowerCase(Locale.ROOT);
+        }
+    }
+
+    public enum FilteringPolicy {
+        EXCLUDE,
+        INCLUDE;
+
+        @Override
+        public String toString() {
+            return name().toLowerCase(Locale.ROOT);
+        }
+    }
+
+    public enum FilteringRuleRule {
```

Comment:
Nitpick: Can we come up with a name that doesn't have duplication? Maybe something like `FilteringRuleCondition`?